### PR TITLE
support `-html` flag to render output as HTML

### PR DESCRIPTION
**Pull request by mdb**
This addresses issue #69 to support HTML output.

This also...

- updates the README usage docs
- validates that multiple, conflicting format flags are not provided
- fixes a bug in the tests to properly validated expected errors invoking `tf-summarize`
- improves the `tf-summarize` tests to exercise the tool's ability to summarize changes to outputs (as well as resources)
- improves `make help` to output documentation for all Makefile targets 
- improve test accuracy via `-count` flag, protecting against false positives in `go test` runs

Seed plant for the future: Ultimately, it might be preferable to support a single `-format="json|md|html|table"` flag, though I was hesitant to do so in this PR and break `tf-summarize`'s existing usage in a non-backwards-compatible fashion.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
+		--tty \
+		--volume $(shell pwd):/src \
+		--workdir /src/example \
+		--entrypoint /bin/sh \
+		hashicorp/terraform:$(1) \
+			-c \
+				"terraform init && \
+				terraform plan -out tfplan && \
+				terraform show -json tfplan > tfplan.json"
+endef
```
Comment (mdb) [inline Makefile:15]: This is a NOOP change moving this macro higher in the Makefile, for formatting purposes.

File: Makefile [hunk 2]
```diff
@@ -13,28 +27,14 @@ build: ## build the binary
 install: build ## build and install to /usr/local/bin/
 	cp $(EXECUTABLE_NAME) /usr/local/bin/$(EXECUTABLE_NAME)
 
-test: lint
-	go test ./...
+test: lint ## go test
+	go test ./... -count=1
```
Comment (mdb) [inline Makefile:31]: ☝️ This ensures against false positives running the tests in between code or testdata changes.

Comment (mdb) [reply Makefile:31]: Note that I also added code comments to all targets within the `Makefile`, such that `make help` documents all targets.

File: README.md [hunk 3]
```diff
@@ -110,10 +110,16 @@ Usage of tf-summarize [args] [tf-plan.json|tfplan]
...
+        [Optional] print changes in html format
+  -json
+        [Optional] print changes in json format
+  -md
+        [Optional, used only with table view] output table as markdown
   -out string
         [Optional] write output to file
   -separate-tree
-        [Optional] print changes in tree format for each add/delete/change/recreate changes
+        [Optional] print changes in tree format for add/delete/change/recreate changes
```
Comment (mdb) [inline README.md:122]: ☝️ The README had gone a bit stale. This fixes that!

File: example/github/main.tf [hunk 4]
```diff
@@ -20,4 +20,8 @@ resource "github_branch" "main" {
 resource "github_branch" "development" {
   repository = github_repository.repository.name
   branch     = "development"
-}
\ No newline at end of file
+}
+
+output "repository_name" {
+  value = github_repository.repository.name
+}
```
Comment (mdb) [inline example/github/main.tf:27]: Previously, the tests did not exercise `tf-summarize`'s ability to summarize output changes. Now they do!

File: writer/html.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import (
+	"io"
+	"path"
+	"text/template"
+
+	"github.com/dineshba/tf-summarize/terraformstate"
+)
+
+// HTMLWriter is a Writer that writes HTML.
+type HTMLWriter struct {
```
Comment (mdb) [inline writer/html.go:12]: This `HTMLWriter` is the meat of the new functionality added by this PR.

File: writer/table.go [hunk 6]
```diff
@@ -44,7 +44,7 @@ func (t TableWriter) Write(writer io.Writer) error {
 	table.Render()
 
 	// Disable the Output Summary if there are no outputs to display
-	if len(t.outputChanges["add"]) > 0 || len(t.outputChanges["delete"]) > 0 || len(t.outputChanges["update"]) > 0 {
```
Comment (mdb) [inline writer/table.go:47]: This logic was a bit brittle and redundant. Now it's moved to a `hasOutputChanges` helper function, which is agnostic to whether the changes are `add`, `delete`, or `update` changes; it simply returns true or false if there are _any_ non-empty `map[string][]string` output changes.

**Comment by mdb** [comment]
👋 Hi @dineshba - I'm curious to hear your thoughts about this PR!

File: writer/table.go (hunk 6 above)
Comment (dineshba) [reply writer/table.go:47]: I am not very sure if there can be any value other than add/delete/update.

**Review by dineshba — APPROVED** [review]

**Pull request merged**